sample-app/src/main/java/com/sinch/sample/conversation/messages/Send.java

```diff
@@ -34,32 +39,70 @@ public void run() {
 
     LOGGER.info("Send message with Conversation");
 
-    var parameters =
-        SendMessageRequest.<TextMessage>builder()
-            .setAppId(conversationAppId)
-            .setMessage(
-                AppMessage.<TextMessage>builder()
-                    .setMessage(
-                        TextMessage.builder()
-                            .setText("[Java SDK: Conversation Message] Sample text message")
-                            .build())
-                    .setAgent(
-                        Agent.builder()
-                            .setType(AgentType.HUMAN)
-                            .setDisplayName("Agent Name")
-                            .build())
-                    .build())
-            .setRecipient(
-                ChannelRecipientIdentities.of(
-                    ChannelRecipientIdentity.builder()
-                        .setChannel(ConversationChannel.SMS)
-                        .setIdentity(phoneNumber)
-                        .build()))
-            .setTtl(25)
-            .build();
+    var parameters = createRCSSendMessage();
 
-    var result = service.sendTextMessage(parameters);
+    var result = service.sendMessage(parameters);
 
     LOGGER.info("Response: " + result);
   }
+
+  SendMessageRequest<?> createRCSSendMessage() {
+    var textMessage =
+        TextMessage.builder()
+            .setText("[Java SDK: Conversation Message] Please select a phone number to be called")
+            .build();
+
+    var choices = new ArrayList<Choice<?>>();
+    for (int i = 0; i < 4; i++) {
+      choices.add(
+          ChoiceCallMessage.builder()
+              .setMessage(
+                  CallMessage.builder()
+                      .setTitle("Phone call choice " + (i + 1))
+                      .setPhoneNumber(phoneNumber)
+                      .build())
+              .setPostbackData("postback call_message data value " + (i + 1))
+              .build());
+    }
+
+    return SendMessageRequest.<ChoiceMessage>builder()
+        .setAppId(conversationAppId)
+        .setMessage(
+            AppMessageWithExtensions.<ChoiceMessage>builder()
+                .setMessage(
+                    ChoiceMessage.builder().setChoices(choices).setTextMessage(textMessage).build())
```

Comment:
Nit: can this be formatted in the following way for better clarity, or the automatic formatter will reset it anyway?
```suggestion
ChoiceMessage.builder()
    .setChoices(choices)
    .setTextMessage(textMessage)
    .build())
```

Reply:
Yes: formatter will reset (and I agree I had preferred too CRLF usage in this context)